Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererArrayBuilder.cs

```diff
@@ -63,14 +64,40 @@ public BocColumnRenderer[] CreateColumnRenderers ()
 
       PrepareSorting(sortingDirections, sortingOrder);
 
-      var firstValueColumnRendered = false;
       var bocColumnRenderers = new List<BocColumnRenderer>(_columnDefinitions.Count);
       var visibleColumnIndex = GetInitialVisibleColumnIndex();
+
+      BocColumnDefinition? iconColumn = null;
+      if (EnableIcon)
+      {
+        var enableIconOnFirstPossibleColumn = !_columnDefinitions.Any(cd => cd is IBocColumnDefinitionWithRowIconSupport { RowIconMode: RowIconMode.Preferred });
+        iconColumn = _columnDefinitions.FirstOrDefault(
+            f =>
+            {
+              if (enableIconOnFirstPossibleColumn)
+              {
+                return f is IBocColumnDefinitionWithRowIconSupport { RowIconMode: RowIconMode.Automatic };
+              }
+              else
+              {
+                return f is IBocColumnDefinitionWithRowIconSupport { RowIconMode: RowIconMode.Preferred };
+              }
+            });
+      }
+
+      var lookup = _columnDefinitions
+          .OfType<IBocColumnDefinitionWithRowIconSupport>()
+          .Select((cd, i) => (cd, i))
+          .ToLookup(x => x.cd.RowIconMode, x => x.i);
+
+      var resultIndex = lookup[RowIconMode.Preferred].OrderBy(i => i).Take(1).Concat(lookup[RowIconMode.Automatic].OrderBy(i => i).Take(1)).FirstOrDefault();
```

Comment:
Did a quick check now (next appointment is looming)

I don't see where `enableIconOnFirstPossibleColumn` is used in the code? Is that an artifact? I would drop line 73-86

line 88 - 93
That's the one I know and looks okay. I'm wondering if it would be easier / have equivalent perofrmance if we did this instead:
var iconCOlumn  = _cols.OfType<IBocColumnDefinitionWithRowIconSupport>().FirstOrDefault(Automatic) ?? _cols.OfType<IBocColumnDefinitionWithRowIconSupport>().FirstOrDefault(Preferred)

I'm fine with either version, it's just a gut feeling about the option I just pasted so I want to make sure it's part of the concious choice :)

Reply:
And yes, this is a duplicate to compare